crates/turbopack-core/src/resolve/pattern.rs

```diff
@@ -11,6 +12,22 @@ use turbo_tasks_fs::{
     DirectoryContent, DirectoryEntry, FileSystemEntryType, FileSystemPathVc, LinkContent, LinkType,
 };
 
+#[turbo_tasks::value(transparent)]
+pub struct QueryMap(#[turbo_tasks(trace_ignore)] Option<IndexMap<String, String>>);
+
+#[turbo_tasks::value_impl]
+impl QueryMapVc {
+    #[turbo_tasks::function]
+    pub fn new() -> Self {
+        Self::cell(Some(IndexMap::new()))
+    }
```

Comment:
Why create an empty query that can't be further modified? Shouldn't this be the same as the `none` case, to avoid allocating the `IndexMap`?